packages/parser/src/parse.ts

```diff
@@ -0,0 +1,43 @@
+import type { LangiumParser } from 'langium';
+
+import type { Info } from './index.js';
+import { createInfoServices } from './language/index.js';
+import type { ILexingError, IRecognitionException } from './language/chevrotainWrapper.js';
+
+type Diagram = Info;
+
+const parsers: Record<string, LangiumParser> = {};
+
+const initializers = {
+  info: () => {
+    // Will have to make parse async to use this. Can try later...
+    // const { createInfoServices } = await import('./language/info/index.js');
+    const parser = createInfoServices().Info.parser.LangiumParser;
+    parsers['info'] = parser;
+  },
+} as const;
+
+export const parse = <T extends Diagram>(
+  diagramType: keyof typeof initializers,
+  text: string
+): T => {
+  const initializer = initializers[diagramType];
+  if (!initializer) {
+    throw new Error(`Unknown diagram type: ${diagramType}`);
+  }
+  if (!parsers[diagramType]) {
+    initializer();
+  }
+  const parser = parsers[diagramType];
+  const { lexerErrors, parserErrors, value } = parser.parse<T>(text);
+  if (lexerErrors.length > 0 || parserErrors.length > 0) {
+    throw new MermaidParseError(lexerErrors, parserErrors);
+  }
+  return value;
+};
+
+export class MermaidParseError extends Error {
+  constructor(lexerErrors: ILexingError[], parserErrors: IRecognitionException[]) {
+    super(`Parsing failed: ${lexerErrors} ${parserErrors}`);
```

Comment:
Attach the actual errors to the error object as fields. Stringifying might not be the best option.

Reply:
I meant making the `lexerErrors` and `parserError` as public properties on the Error object, so we can access them in the mermaid project as actual objects.